packages/blocks/src/api/parser/fix-custom-classname.js

```diff
@@ -10,6 +10,13 @@ import { hasBlockSupport } from '../registration';
 import { getSaveContent } from '../serializer';
 import { parseWithAttributeSchema } from './get-block-attributes';
 
+const CLASS_ATTR_SCHEMA = {
+	type: 'string',
+	source: 'attribute',
+	selector: '[data-custom-class-name] > *',
+	attribute: 'class',
+};
```

Comment:
Should we do the same for the "style" attribute which is also a common one shared by a lot of blocks?

Reply:
I don't see the `style` attribute being read with this method anywhere. It's usually stored in the block comment (`<!-- wp:heading {"style"} -->`), not in the block HTML markup, is that right? Then the code path is different and doesn't involve HTML parsing and matching.

Reply:
Makes sense, so we only care about "memoizing" attribute definitions if the attribute is parsed from content. I wonder if there's a way to surface this for block authors. In documentation at least.

Reply:
For block authors, I think this matters only in rare extreme cases. Like when the block has many deprecated versions with shared attribute definitions. For example, the `core/gallery` block has six deprecated versions, and they share definitions for attributes like `img[data-full-url]`. But deduplicating them would have very little impact.

Where would be the best place to document this? We can do that anyway.

Reply:
Indeed, since most block use block.json and don't add anything "common" (rarely), it's not that important. I'm also not sure how to document this, so not important.